src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1,9 +1,93 @@
 package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
 
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.UUID;
+import ma.glasnost.orika.CustomConverter;
+import ma.glasnost.orika.MapperFacade;
+import ma.glasnost.orika.MapperFactory;
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.converter.ConverterFactory;
+import ma.glasnost.orika.impl.DefaultMapperFactory;
+import ma.glasnost.orika.metadata.Type;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 import org.springframework.validation.annotation.Validated;
+import org.springframework.web.server.ResponseStatusException;
+import uk.gov.ons.ctp.common.time.DateTimeUtil;
+import uk.gov.ons.ctp.integration.contactcentresvc.client.caseservice.CaseServiceClientServiceImpl;
+import uk.gov.ons.ctp.integration.contactcentresvc.client.caseservice.model.CaseContainerDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.CaseRequestDTO;
 import uk.gov.ons.ctp.integration.contactcentresvc.service.CaseService;
+import uk.gov.ons.ctp.integration.contactcentresvc.utility.Constants;
 
 @Service
 @Validated()
-public class CaseServiceImpl implements CaseService {}
+public class CaseServiceImpl implements CaseService {
+  private static final Logger log = LoggerFactory.getLogger(CaseServiceImpl.class);
+
+  private MapperFacade caseDTOMapper;
+
+  @Autowired private CaseServiceClientServiceImpl caseServiceClient;
+
+  private static class DateMapper extends CustomConverter<String, Date> {
+    private SimpleDateFormat caseDateTimeFormatter =
+        new SimpleDateFormat(DateTimeUtil.DATE_FORMAT_IN_JSON);
+
+    @Override
+    public Date convert(
+        String dateAsText, Type<? extends Date> destinationType, MappingContext mappingContext) {
+      try {
+        return caseDateTimeFormatter.parse(dateAsText);
+      } catch (ParseException e) {
+        throw new RuntimeException("Failed to parse date: " + dateAsText);
+      }
+    }
+  }
+
+  public CaseServiceImpl() {
+
```

Comment:
Would rather see the mapper config outside the service - the pattern we have used in the past is to register mapper config in a BeanMapper class in the project root package - this project has one, although it is currently empty
Prior art from our old CaseService - 

  protected final void configure(final MapperFactory factory) {

    factory.classMap(Case.class, CaseDTO.class).byDefault().register();

    factory.classMap(Case.class, CaseDetailsDTO.class).byDefault().register();

    factory
        .classMap(CaseGroup.class, CaseGroupDTO.class)
        .field("status", "caseGroupStatus")
        .byDefault()
        .register();

    factory.classMap(CaseEvent.class, CaseEventDTO.class).byDefault().register();

    factory
        .classMap(Category.class, CategoryDTO.class)
        .field("categoryName", "name")
        .byDefault()
        .register();

    factory.classMap(Response.class, ResponseDTO.class).byDefault().register();
  }

That one component can then be autowired into any class in the project needing to perform an orika mapping.

As it stands this mapping knowledge is only local to the CaseServiceImpl

Reply:
Initialisation code moved to the mapping class.